backend/corpora/common/utils/dl_sources/url.py

```diff
@@ -37,6 +37,16 @@ def _get_key(self, headers: dict, key: str) -> str:
                 f"{self.__class__.__name__}:URL({self.url}) failed request. '{key}' not present in the header."
             )
 
+    def _get_key_with_fallback(self, headers: dict, key: str, fallback_key: str) -> str:
```

Comment:
Why not pass in a list and put the keys in priority order. I can see this functions getting overly verbose with arguments at this rate.

Reply:
This is a hotfix, I prefer a solution which is explicit and contains the least amount of logic. I'm happy to refactor it later and add some unit tests, but that should be done on the main branch.